### PR TITLE
Do not crash in "too few arguments" error with anon argument

### PR DESCRIPTION
**Pull request by sixolet**
This should fix https://github.com/python/mypy/issues/3312

### CODE REVIEW

File: test-data/unit/check-functions.test [hunk 1]
```diff
@@ -2039,3 +2039,8 @@ l: List[int] = make_list()
 
 bad = make_list()  # E: Need type annotation for variable
 [builtins fixtures/list.pyi]
+
+[case testAnonymousArgumentError]
+
```
Comment (gvanrossum) [inline test-data/unit/check-functions.test]: Don't need this blank line.

File: mypy/messages.py [hunk 2]
```diff
@@ -575,7 +575,7 @@ def too_few_arguments(self, callee: CallableType, context: Context,
                 msg = 'Missing positional argument'
             else:
                 msg = 'Missing positional arguments'
-            if callee.name and diff:
+            if callee.name and [d for d in diff if d is not None]:
```
Comment (gvanrossum) [inline mypy/messages.py]: This could just be `if callee.name and any(diff):` ... Or if that's too cryptic use `any(d is not None for d in diff)`.

Comment (sixolet) [inline mypy/messages.py]: Heh.  Should be `all`, thanks for the collaborative catch.

**Comment by sixolet** [comment]
... wow, who'd have thunk I'd get something so simple wrong so many times.

**Comment by gvanrossum** [comment]
> ... wow, who'd have thunk I'd get something so simple wrong so many times.

It's the release pressure. Happens to me all the time. Also, this logic *is* pretty subtle. And thanks for fixing this under pressure!

**Pull request merged**